js/product-details.js

```diff
@@ -0,0 +1,161 @@
+var xmlhttp = new XMLHttpRequest();
+xmlhttp.onreadystatechange = function () {
+  if (this.readyState == 4 && this.status == 200) {
+    var data = JSON.parse(this.responseText);
+    var drink = data[productCategory][productId].additionalDescription;
+    for (key in drink) {
+      if (drink.hasOwnProperty(key)) {
+        if (!drink[key]) {
+        } else {
+          buildDetailButton(drink);
+          bildDetailvaluecontainers(drink);
+        }
+      }
+    }
+    addClickEventListenerToButtons(drink);
+    activateFirstButton()
+  }
+};
+xmlhttp.open("GET", "../jsons/" + productCategory + ".json", true);
+xmlhttp.send();
+
+const activateFirstButton = function () {
+  document.getElementsByClassName("details-description")[0].classList.add("details-button-active--row");
+  document.getElementsByClassName("product-details-value--row")[0].classList.toggle("product-details-value--row--none");
+}
+
+const buildDetailButton = function (drink) {
+  var productDetails = document.getElementsByClassName("product-details")[0];
+  let detailButton = document.createElement("div");
+  detailButton.innerHTML = key;
+  detailButton.classList.add("details-" + key);
+  detailButton.classList.add("details-button");
+  productDetails.appendChild(detailButton);
+};
+
+const bildDetailvaluecontainers = function (drink) {
+  var containerColumn = document.getElementsByClassName("product-details")[0];
+  var containerRow = document.getElementsByClassName(
+    "product-details-value"
+  )[0];
+  if (typeof drink[key] == "object" && !Array.isArray(drink[key])) {
+    if (drink.hasOwnProperty(key)) {
+      ifValueObject(drink[key], containerColumn, containerRow)
+    }
+  } else if (Array.isArray(drink[key])) {
+    ifValueArray(drink[key], containerColumn, containerRow)
+  } else {
+    ifValueString(drink[key], containerColumn, containerRow);
+  }
+};
+
+const ifValueObject = function (value, containerColumn, containerRow) {
```

Comment:
function that starts from 'if' means that this function check something and return boolean value true or false
for example ifButtonActive, ifUserLoggedIn, ifSashaCanNameFunction)))
and on function name I see that function checks if some value passed to it is an Object
but function doesn't do this. it event doesn't return anything
maybe something like 'showProductInfo'? or like that
think about it and set name that will definitely describe what function does.